host_applications/linux/apps/raspicam/RaspiVid.c

```diff
@@ -806,6 +844,40 @@ static int parse_cmdline(int argc, const char **argv, RASPIVID_STATE *state)
          break;
       }
 
+      case CommandRaw:  // output filename
+      {
+         state->raw_output = 1;
+         state->raw_output_fmt = RAW_OUTPUT_FMT_YUV;
+         int len = strlen(argv[i + 1]);
+         if (len)
+         {
+            state->raw_filename = malloc(len + 1);
+            vcos_assert(state->raw_filename);
+            if (state->raw_filename)
+               strncpy(state->raw_filename, argv[i + 1], len+1);
+            i++;
+         }
+         else
+            valid = 0;
+         break;
+      }
+
+      case CommandRawFormat:
+      {
+         state->raw_output_fmt = raspicli_map_xref(argv[i + 1], raw_output_fmt_map, raw_output_fmt_map_size);
+         i++;
+
+         if (state->raw_output_fmt == -1)
+            state->raw_output_fmt = RAW_OUTPUT_FMT_YUV;
```

Comment:
So if they enter "--raw-format gibberish" they'll get YUV? I'd have thought we really want to reject that instead.